File: web/web-frontend/src/app/model/runner/tree/runner-parametrized-test.node.ts
import { Path } from "../../infrastructure/path/path.model";
import { RunnerTreeDeserializationUtil } from "./util/runner-tree-deserialization.util";
import { RunnerTestOrFeatureNode } from "./runner-test-or-feature-node.model";
import { RunnerStepNode } from "./runner-step-node.model";
import {Serializable} from "../../infrastructure/serializable.model";
import {RunnerScenarioNode} from "./runner-scenario-node.model";

export class RunnerParametrizedTestNode implements RunnerTestOrFeatureNode, Serializable<RunnerParametrizedTestNode> {
    id: string;
    path: Path;
    name: string;
    children: Array<RunnerScenarioNode> = [];

    deserialize(input: Object): RunnerParametrizedTestNode {
        this.id = input["id"];
        this.name = input["name"];
        this.path = Path.deserialize(input["path"]);

        this.children = RunnerTreeDeserializationUtil.deserializeRunnerScenariosNodes(input["children"] || []);

        return this;
    }

    serialize(): string {
        throw Error("method not implemented")
    }
}

File: model/src/main/kotlin/com/testerum/model/runner/old_tree/RunnerParametrizedTestNode.kt
package com.testerum.model.runner.old_tree

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.testerum.model.infrastructure.path.Path

data class RunnerParametrizedTestNode @JsonCreator constructor(@JsonProperty("id") override val id: String,
                                                               @JsonProperty("path") override val path: Path,
                                                               @JsonProperty("name") val name: String,
                                                               @JsonProperty("children") val children: List<RunnerScenarioNode>): RunnerTestOrFeatureNode
